### PR TITLE
DOC add an example with sklearn.pipeline.Pipeline 

### PR DESCRIPTION
**Pull request by romanlutz**
Adds an example that uses `sklearn.pipeline.Pipeline` to show that Fairlearn works with estimators that are made up of various steps. This is especially useful since we previously weren't compatible with `Pipeline`.

Update: refactored the following out into #918. ~~Additionally, I'm renaming `interpolation_dict` to `threshold_interpolation_` since we never liked that name to begin with (see #614) and it's really the information about how to interpolate between thresholds to get your predictions. If people dislike this name I can undo this, though. Since `InterpolatedThresholder` isn't a class that's actually visible in our API docs I don't think this constitutes a breaking change. However, we may want to consider making it public by explicitly mentioning it in the postprocessing module's `__init__.py`.~~

This is also related to #665 since it represents a first step towards replacing usage of `get_dummies`.
Furthermore, #614 will need to be updated following this PR.

### CODE REVIEW

File: examples/plot_mitigation_pipeline.py
Comment (adrinjalali) [inline examples/plot_mitigation_pipeline.py:28]: I'd add a section here:

```suggestion
# %%
```

File: examples/plot_mitigation_pipeline.py
Comment (adrinjalali) [inline examples/plot_mitigation_pipeline.py:43]: same here (with a one sentence saying what the sections do)

File: fairlearn/postprocessing/_interpolated_thresholder.py
Comment (adrinjalali) [inline fairlearn/postprocessing/_interpolated_thresholder.py:83]: this looks like a meta-estimator to me, so I'd rather it follow estimator API and have this as `threshold_interpolation` instead of the one with a trailing underscore.

**Review by adrinjalali — APPROVED** [review]
I'm happy with the PR, but I think this is actually two separate PRs?

**Comment by adrinjalali** [comment]
Also, to me this is a DOC PR rather than a FEAT

File: fairlearn/postprocessing/_interpolated_thresholder.py
Comment (romanlutz) [reply fairlearn/postprocessing/_interpolated_thresholder.py:83]: It is indeed a meta-estimator. Changed it in the latest commit (which goes to a separate PR), but I have to say I wasn't aware of different rules for meta-estimators vs estimators. If you happen to have a reference I'd love to educate myself 🙂

**Comment by romanlutz** [comment]
Refactored the `InterpolatedThresholder` changes out into #918. Let's merge #918 first and then this one (which includes the same changes and builds on them).

File: fairlearn/postprocessing/_interpolated_thresholder.py
Comment (adrinjalali) [reply fairlearn/postprocessing/_interpolated_thresholder.py]: I meant "it's an estimator because it's a meta-estimator." All meta-estimators are estimators, and there isn't much to it other than: 

https://github.com/scikit-learn/scikit-learn/blob/45bb9ab916351c95a6d15a5cb06469036a0707cf/sklearn/base.py#L810-L813

Comment (adrinjalali) [reply fairlearn/postprocessing/_interpolated_thresholder.py]: In more advanced cases, there are things such as _estimator tag delegation_ and passing `fit_params` and `score_params` to the underlying estimator, etc, but that can come next.

**Review by adrinjalali — APPROVED** [review]
LGTM, good to merge once the other PR is merged.

**Comment by romanlutz** [comment]
Looks like the other PR may have a hard(er) time so I'll just remove those changes from here and will update the other one accordingly. That way we can merge this one asap.

File: examples/plot_mitigation_pipeline.py
Comment (riedgar-ms) [inline examples/plot_mitigation_pipeline.py:101]: No GridSearch?

Comment (romanlutz) [reply examples/plot_mitigation_pipeline.py:101]: I didn't really want to show each and every mitigation technique, but felt like it makes sense to have one of postprocessing and reductions each. I expect we'll have, say, 20 mitigation techniques at some point, and we can't have all of them everywhere. Does that make sense?

Comment (riedgar-ms) [reply examples/plot_mitigation_pipeline.py:101]: OK

**Review by riedgar-ms — APPROVED** [review]

**Comment by romanlutz** [comment]
Looks like CircleCI refuses to schedule a build for this pipeline. I've even tried adding another commit, but it didn't trigger it. To prove that it does indeed work I've run this locally. The changes are all on a single page, so it should be easy to check the screenshots:

image

image

image

The warning about the pipeline will be fixed once scikit-learn provides a way to properly check for whether a pipeline is fitted (currently in PR).

Since these look just like they should I'll go ahead and merge this since we have the necessary approvals and this has been sitting around for a few days.

**Comment by romanlutz** [comment]
Voila: https://fairlearn.org/main/auto_examples/plot_mitigation_pipeline.html#sphx-glr-auto-examples-plot-mitigation-pipeline-py

**Comment by adrinjalali** [comment]
Thanks @romanlutz

**Pull request merged**
